pkgutil/pkgutil.go

```diff
@@ -0,0 +1,66 @@
+package pkgutil
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	zglob "github.com/mattn/go-zglob"
+)
+
+// LongestCommonPrefix returns the longest prefix of all strings the argument
+// slice. If the slice is empty the empty string is returned
+func LongestCommonPrefix(strs []string) string {
+	if len(strs) == 0 {
+		return ""
+	}
+	lcp := strs[0]
+	for _, str := range strs {
+		lcp = strlcp(lcp, str)
+	}
+	return lcp
+}
+
+func strlcp(a string, b string) string {
+	var min int
+	if len(a) > len(b) {
+		min = len(b)
+	} else {
+		min = len(a)
+	}
+	for i := 0; i < min; i++ {
+		if a[i] != b[i] {
+			return a[0:i]
+		}
+	}
+	return a[0:min]
+}
+
+// Glob returns a map with source file path as keys and destination as values.
+// First the longest common prefix (lcp) of all globbed files is found. The destination
+// for each globbed file is then dst joined with src with the lcp trimmed off.
+func Glob(glob, dst string) (map[string]string, error) {
+	matches, err := zglob.Glob(glob)
+	if err != nil {
+		return nil, err
+	}
+	if len(matches) == 0 {
+		return nil, fmt.Errorf("no files matched by: %s", glob)
+	}
+	files := make(map[string]string)
+	prefix := LongestCommonPrefix(matches)
```

Comment:
i don't get why we need this..

Reply:
oh, nevermind, just realized it...

Reply:
to be clear: by removing the longest prefix we enable the user to have the greatest control over the resulting destination path.